wammu.1
.TH wammu 1 "2005-01-24" "Mobile phone manager"

.SH NAME
wammu \- program for managing entries in your mobile phone

.SH SYNOPSIS
.B wammu
.RI [ options ]
.br

.SH DESCRIPTION
This manual page explains the
.B wammu
program. This program is graphical interface for gammu.

.SH OPTIONS
These programs follow the usual GNU command line syntax, with long
options starting with two dashes (`-').
A summary of options is included below.
.TP
.B \-h, \-\-help
Show summary of options.
.TP
.B \-v, \-\-version
Show version of program.
.TP
.B \-l, \-\-local\-locales
Use locales from current directory rather than system ones. This is
mostly useful for development or when running from unpacked tarball
without installation.
.TP
.B \-i, \-\-info
Prints connection settings and tries to connect the phone and display 
some basic information about it. This does not use GUI.
.TP
.B \-d, \-\-debug
Enables printing of debug information to stderr. Work for both GUI 
and \-\-info.

.SH LICENSE
This program is licensed under GNU/GPL version 3 or later.

.SH REPORTING BUGS
There are definitely many bugs, reporting to author is welcome. Please include
some useful information when sending bug reports (eg. exception you received
and debug output). Please submit your reports to <https://github.com/gammu/wammu/issues/>.

.SH SEE ALSO
More information is available on program website:
<\fIhttps://wammu.eu/\fR>.

gammu(1)

wammu-configure(1)

.SH AUTHOR
Michal Cihar <\fImichal@cihar.com\fR>
.SH COPYRIGHT
Copyright \(co 2003 - 2008 Michal Cihar <\fImichal@cihar.com\fR>


NEWS
NEWS
====

Major changes for each release are described here. Detailed changes between
two versions can be retrieved from Subversion.

0.26
==== (2008-??-??)

* Various fixes to make Windows version back to usable state.

0.25
==== (2007-12-19)

* Fall back to Add methods if Set is not supported.
* Fixed phone number validation.
* Fixed crash when org.freedesktop.Notifications service is not
  provided.
* Correctly report why guided wizard can not connect to phone.
* New navigation icons, thanks to David Corrales.

0.24
==== (2007-11-13)

* Notifies user about incoming calls using D-Bus.
* Remember default type of created entry (eg. phone/SIM memory).
* Newly created entries contain some reasonable fields by default.
* Fixed validation of editor fields.
* Improvements in guided wizard to make it easier to use.
* Clever searching over Bluetooth based on manufacturer.
* Searching can be really limited to some connection types.

0.23
==== (2007-10-05)

* Fixed phone number prefix settings.
* Customizable format for displaying name.
* Fixed crash when reporting information to phone database.

0.22
==== (2007-09-11)

* Fixed errors in locales and applied stricter checking.
* Fixed various crashes.

0.21
==== (2007-06-20)

* Tell user if he needs to be member of some group while searching.
* Avoid blocking opening of a browser.

0.20
==== (2007-05-16)

* Improve locales behavior (and require wxPython with Unicode).
* Show time in call listings.
* Fixed infinite loop in searching.
* Fixed displaying of messages with special chars.
* Fixed locale detection on system without configured locales.
* Use different approach for detecting home directory on Windows.
* Now requires PyWin32 on Windows.

0.19
==== (2007-03-11)

* Improved SMS recipient list handling.
* Can save and load SMS recipient list.
* Fix crash on Windows when username contains non-ascii chars.
* Fix Windows binary crashes.
* Support for sending file to phone.
* Support for task bar icon (thanks to Rene Peters).
* Fixed debug log on Windows.
* Fixed error handler on Windows.

0.18
==== (2007-02-24)

* Better setup.py to use more of distutils.
* .mo locales are generated on build and not included in VCS.
* Show only existing devices on Windows (this requires pywin32).
* Now requires wxPython at least 2.6.2.
* Support for reporting in Gammu Phone Database.
* Automatically asks user for feedback after month of usage.
* GUI more follows HIG on target platform (button placement).
* Add wildcard search possibility.
* Use gammu configuration for phone connection instead of own.

0.17
==== (2006-10-19)

* Fix searching.
* Remember column sorting.

0.16
==== (2006-09-21)

* Include platform in crash log.
* Fix SMS composer on Windows.
* Fix adding part to SMS after removing all.
* Fix editor on Windows.
* Now requires wxWidgets 2.6.

0.15
==== (2006-09-03)

* Language selection for Windows (based on GTK+ 2 ISS code).
* Finnish translation from Rosetta.
* Searching can be done plain text and not only regexp.
* Phone number matching now works even for national numbers.
* Fixed usability with inversed themes.

0.14
==== (2006-08-05)

* Automatically append backup extension.
* PyBluez is now default for Bluetooth devices lookup.
* Fixed mail export for non English locales.
* Fixed installation on Windows.
* Added support for py2exe for distributing Windows executable.
* Added script for creating Wammu installer on Windows (using Inno Setup).

0.13
==== (2006-03-24)

* Possibility to export SMS to maildir, mailbox and IMAP account.
* More user friendly debug logging.
* Many new translations, thanks to Rosetta.
* Improved Bluetooth lookup - now uses either PyBluez or btctl (part of GNOME
  Bluetooth).

0.12
==== (2006-01-08)

* More user friendly reporting of some common errors.
* Many bug fixes.

0.11
==== (2005-12-24)

New features:

* SMS backups and import support.

0.10
==== (2005-08-07)

Bug fixes:

* Fixed wrong implementation of not synced items for SMS.
* Some minor display errors were fixed.
* Application icon is now used correctly.

0.9
=== (2005-07-24)

New features:

* Searching in entries.
* Reading backups without phone connected, can work as backup viewer.
* New bug reporting dialog, added bugid to allow easily search for
  similar bugs.
* Support for searching Bluetooth devices using gnomebt python module.
* Better searching for Nokia phones.

Bug fixes:

* Fixed some issues with debug wxPython, probably still all not solved.

0.8
=== (2004-09-17)

New features:

* New settings dialog.
* Better date editing control.

Bug fixes:

* Should work with both wxPython 2.5 and 2.4.

Known issues:

* Calendar editor does not work in wxPython 2.5/Gtk 2, you need to enter
  values manually (this is wxWidgets issue and has been fixed there).

0.7
=== (2004-09-06)

New features:

* Signal and battery state continuous monitoring.
* Support for sending messages to multiple recipients.

Bug fixes:

* Should work with wxPython 2.5.

0.6
=== (2004-03-03)

New features:

* Can expand "compressed" SMSes - TextCOMPRESSIONlikeTHIS.
* Allow direct calls/send message for contacts.

Bug fixes:

* Lots of charset issues fixed.

0.5.1
===== (2004-01-23)

* Fixed typo in Wammu/Utils.py.

0.5
=== (2004-01-21)

New features:

* Searching for phone.
* Backup and import functionality.
* Deleting of multiple entries at once.
* Reply on message.
* Playback of ring tones using timidity.
* More decoded SMS parts are now recognized (especially Nokia ones were not
  recognized before).
* Unexpected exceptions are handled and reported as bugs.
* Czech localization.

Bug fixes:

* HTML displayer now should display non ascii texts correctly.
* Reads really all SMS.
* Better handling of exceptions when reading data.
* Broken wxListCtrl.GetColumn on wxPython 2.4.1.2 is workarounded.
* Workaround charset conversion problems in wxPython.
* Fixed creating multiple roots in tree control.

0.4
=== (2004-01-06)


wammu.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
#
# Copyright © 2003 - 2017 Michal Čihař <michal@cihar.com>
#
# This file is part of Wammu <https://wammu.eu/>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <https://www.gnu.org/licenses/>.
'''
Wammu - Phone manager
Execution script
'''
from __future__ import unicode_literals
from __future__ import print_function

import os
import sys
import getopt
try:
    import Wammu
except ImportError:
    print('Failed to import Wammu module, trying to fix sys.path!')
    sys.path.append(
        os.path.join(
            os.path.dirname(__file__),
            '..', 'lib', 'python2.7', 'site-packages'
        )
    )
    sys.path.append(
        os.path.join(
            os.path.dirname(__file__),
            '..', 'lib', 'python2.7', 'dist-packages'
        )
    )
    import Wammu
import Wammu.Locales
from Wammu.Locales import ugettext as _

# Disable warning about missing files
# This can be caused by attempt to import Python modules, which don't
# have all DLLs satisfied.
if sys.platform.startswith('win'):
    import win32api
    import win32con
    win32api.SetErrorMode(win32con.SEM_NOOPENFILEERRORBOX)


def version():
    '''
    Displays version information.
    '''
    print(_('Wammu - Windowed Gammu version %s') % Wammu.__version__)


def usage():
    '''
    Shows program usage.
    '''
    version()
    print(_('Usage: %s [OPTION...]' % os.path.basename(__file__)))
    print()
    print(_('Options:'))
    print('%-20s ... %s' % (
        '-h/--help',
        _('show this help')
    ))
    print('%-20s ... %s' % (
        '-v/--version',
        _('show program version')
    ))
    print('%-20s ... %s' % (
        '-l/--local-locales',
        _('force using of locales from current directory rather than system ones')
    ))
    print('%-20s ... %s' % (
        '-i/--info',
        _('prints connection settings and tries to connect the phone')
    ))
    print('%-20s ... %s' % (
        '-d/--debug',
        _('enables debug output to stderr')
    ))
    print()


def info():
    '''
    Displays configuration summary and tries to connect to phone.
    '''
    import Wammu.WammuSettings
    import Wammu.Utils
    import gammu

    settings = Wammu.WammuSettings.WammuConfig()
    section = settings.ReadInt('/Gammu/Section')
    config = settings.gammu.GetConfig(section)
    if config['Connection'] == '' or config['Device'] == '':
        print(_('Wammu is not configured!'))

    cfg = {
        'StartInfo': settings.ReadBool('/Gammu/StartInfo'),
        'UseGlobalDebugFile': True,
        'DebugFile': None,  # Set on other place
        'SyncTime': settings.ReadBool('/Gammu/SyncTime'),
        'Connection': config['Connection'],
        'LockDevice': settings.ReadBool('/Gammu/LockDevice'),
        'DebugLevel': 'textalldate',  # Set on other place
        'Device': config['Device'],
        'Model': config['Model'],
    }

    # Compatibility with old Gammu versions
    cfg = Wammu.Utils.CompatConfig(cfg)

    print(_('Wammu configuration:'))
    print('%-15s: %s' % (_('Connection'), cfg['Connection']))
    print('%-15s: %s' % (_('Model'), cfg['Model']))
    print('%-15s: %s' % (_('Device'), cfg['Device']))
    print(_('Connecting...'))
    if Wammu.debug:
        gammu.SetDebugFile(sys.stderr)
        gammu.SetDebugLevel('textalldate')
    sm = gammu.StateMachine()
    sm.SetConfig(0, cfg)
    sm.Init()
    print(_('Getting phone information...'))
    Manufacturer = sm.GetManufacturer()
    Model = sm.GetModel()
    IMEI = sm.GetIMEI()
    Firmware = sm.GetFirmware()
    Code = sm.GetSecurityStatus()
    print(_('Phone infomation:'))
    print('%-15s: %s' % (_('Manufacturer'), Manufacturer))
    print('%-15s: %s (%s)' % (_('Model'), Model[0], Model[1]))
    print('%-15s: %s' % (_('IMEI'), IMEI))
    print('%-15s: %s' % (_('Firmware'), Firmware[0]))
    if Code is not None:
        print('%-15s: %s' % (_('Requested code'), Code))


def parse_options():
    '''
    Processes program options.
    '''
    try:
        opts, args = getopt.getopt(
            sys.argv[1:],
            'hvlid',
            ['help', 'version', 'local-locales', 'info', 'debug']
        )
    except getopt.GetoptError as val:
        usage()
        print(_('Command line parsing failed with error:'))
        print(val)
        sys.exit(2)

    if len(args) != 0:
        usage()
        print(_('Extra unrecognized parameters passed to program'))
        sys.exit(3)

    do_info = False

    for opt, dummy in opts:
        if opt in ('-l', '--local-locales'):
            Wammu.Locales.UseLocal()
            print(_('Using local built locales!'))
        if opt in ('-h', '--help'):
            usage()
            sys.exit()
        if opt in ('-v', '--version'):
            version()
            sys.exit()
        if opt in ('-i', '--info'):
            do_info = True
        if opt in ('-d', '--debug'):
            Wammu.debug = True

    if do_info:
        info()
        sys.exit()


if __name__ == '__main__':
    Wammu.Locales.Init()
    parse_options()
    # need to be imported after locales are initialised
    import Wammu.App
    Wammu.App.Run()
